### PR TITLE
Discover descriptors on connect

### PR DESCRIPTION
**Pull request by FabioCornelli**
Hi,

first of all thank you for this library! 
it's really helpful for managing BLE across different platforms using kotlin common code.

We are trying to use Kable in our mobile app (android & iOS) and we noticed an issue / unexpected behavior.

It seems that on Apple platforms, after establishing a connection to a peripheral , descriptors are not discovered.

This is causing issues / unexpected behaviors, because on other platform (e.g. Android) all services, characteristics and descriptors are discovered when calling `peripheral.connect()`

For example, imagine a code like this in **commonMain** kotlin code:

```kotlin
val advertisement = scanner.advertisements.first()
val peripheral = coroutineScope.peripheral(advertisement)
peripheral.connect()
peripheral.services?.forEach { service ->
    println("service ${service.serviceUuid}")
    service.characteristics.forEach { characteristic ->
        println("characteristic ${characteristic.characteristicUuid}")
        val descriptors = characteristic
            .descriptors
            .map { descriptor -> peripheral.read(descriptor).decodeToString() }
        println("descriptors -> $descriptors")
    }
}
```

this code will have different behavior and will perform different BLE operation.

On Android it will discover all services, characteristics and read all descriptors.

On iOS it will only discover all services and characteristics, but will not discover nor read any descriptor.

With this PR I've tried to align the behavior on apple platforms.

I've also fixed  `fun readAsNSData(descriptor: Descriptor ): NSData`  because it assumed that descriptor was always in NSData format, but it wasn't actually always the case.

Let me know if this PR seems ok and can be merged or if it needs any modification.
 
Thank you!

### CODE REVIEW

**Comment by twyatt** [comment]
> first of all thank you for this library!

Aww, thanks!

> It seems that on Apple platforms, after establishing a connection to a peripheral , descriptors are not discovered.

For some reason, I thought this was automatic, but if not, then this PR is definitely needed.

> On iOS it will only discover all services and characteristics, but will not discover nor read any descriptor.

Wow, I was not aware of this behavior on iOS. Thanks for pointing this out!

> Let me know if this PR seems ok and can be merged or if it needs any modification.

First off, thanks for raising this issue and creating this PR! 💯
I'll try to find time this week (or this weekend) to review it.

**Comment by twyatt** [comment]
Sorry, didn't find time over the weekend, but I plan to review this in the next day or two.

**Comment by FabioCornelli** [comment]
No worries, in the meantime I've fixed a couple of minor Lint errors reported by CI build and rebased on latest version of the library.

File: core/src/appleMain/kotlin/CBPeripheralCoreBluetoothPeripheral.kt
Comment (twyatt) [inline core/src/appleMain/kotlin/CBPeripheralCoreBluetoothPeripheral.kt:431]: This whole section, despite being very useful, seems out-of-scope of this PR (belongs in a separate PR)?

Additionally, being that these are UUIDs for standard descriptors, it probably makes sense to provide similar functionality for the other platforms as well, right?

File: core/src/appleMain/kotlin/CBPeripheralCoreBluetoothPeripheral.kt
Comment (twyatt) [inline core/src/appleMain/kotlin/CBPeripheralCoreBluetoothPeripheral.kt:277]: Is the `filterIsInstance` so that we don't have to do the cast below?

Despite this being cleaner code, I think I'd still prefer the "unsafe" cast as it avoids an object allocation that `filterIsInstance` incurs.

```suggestion
        // Cast should be safe since `CBPeripheral.services` type is `[CBService]?`, according to:
        // https://developer.apple.com/documentation/corebluetooth/cbperipheral/services
        val discoveredServices = cbPeripheral.services as List<CBService>?
        discoveredServices?.forEach { cbService ->
```

File: core/src/appleMain/kotlin/CBPeripheralCoreBluetoothPeripheral.kt
Comment (twyatt) [inline core/src/appleMain/kotlin/CBPeripheralCoreBluetoothPeripheral.kt:283]:
```suggestion
            // Cast should be safe since `CBService.characteristics` type is `[CBCharacteristic]?`,
            // according to: https://developer.apple.com/documentation/corebluetooth/cbservice/characteristics
            @Suppress("UNCHECKED_CAST")
            val discoveredCharacteristics = cbService.characteristics as List<CBCharacteristic>?
            discoveredCharacteristics?.forEach { cbCharacteristic ->
                connection.execute<DidDiscoverDescriptorsForCharacteristic> {
                    centralManager.discoverDescriptors(cbPeripheral, cbCharacteristic)
```

**Review by twyatt — CHANGES_REQUESTED** [review]
First off, thanks so much for creating this PR! This is a major gap in the iOS implementation, so this PR is super helpful.

Although, I feel like there are 2 changes that are occurring that can be split into 2 PRs:
1. Descriptor discovery (this PR)
2. Improved descriptor read support (creating an additional PR)

_I'm assuming those two changes aren't tightly coupled (and could each function on their own)?_

Other than that, this looks great, I just have some minor changes for this "descriptor discovery" PR.

Thanks again! 💯

File: core/src/appleMain/kotlin/CBPeripheralCoreBluetoothPeripheral.kt
Comment (FabioCornelli) [reply core/src/appleMain/kotlin/CBPeripheralCoreBluetoothPeripheral.kt:431]: Yes you are right, this section was added because I've encountered another issue after I've solved the original descriptor discovery problem.
I'll open a separate PR for this.

> Additionally, being that these are UUIDs for standard descriptors, it probably makes sense to provide similar functionality for the other platforms as well, right?

Yes, I need to verify the behavior on other platforms and maybe is possible to provide common code for this.

File: core/src/appleMain/kotlin/CBPeripheralCoreBluetoothPeripheral.kt
Comment (FabioCornelli) [reply core/src/appleMain/kotlin/CBPeripheralCoreBluetoothPeripheral.kt]: yes I've used filterIsInstance because I needed cbService casted to `CBService ` also in the next code block and didn't want to double cast it.

I'll push an update to the PR with the approach you suggested

File: core/src/appleMain/kotlin/CBPeripheralCoreBluetoothPeripheral.kt
Comment (FabioCornelli) [reply core/src/appleMain/kotlin/CBPeripheralCoreBluetoothPeripheral.kt]: I'll push an update to the PR with the approach you suggested

**Review by twyatt — APPROVED** [review]

**Comment by twyatt** [comment]
I'll merge after we get another approval and I've run a few quick tests.

File: core/src/appleMain/kotlin/CBPeripheralCoreBluetoothPeripheral.kt
Comment (FabioCornelli) [reply core/src/appleMain/kotlin/CBPeripheralCoreBluetoothPeripheral.kt]: I've just opened #706 for this :)

**Review by davidtaylor-juul — APPROVED** [review]
LGTM

**Comment by twyatt** [comment]
Tested working with SensorTag sample app:

<details>
<summary><code>./gradlew runDebugExecutableMacosArm64</code></summary>

```
> Task :app:runDebugExecutableMacosArm64
[I/SensorTag] Searching for SensorTag...
2024-07-02 14:43:24.194 sensortag.kexe[35961:336063] I/Kable/Scanner: Starting scan with native service filtering
[I/SensorTag] Found Advertisement(identifier=6a9bbeca-ec4d-5f75-4de5-2f59e392836f, name=null, rssi=-67, txPower=null)
2024-07-02 14:43:53.663 sensortag.kexe[35961:336063] I/Kable/Scanner: Stopping scan
[I/SensorTag] Configuring auto connector
[I/SensorTag] Received state: Disconnected(null)
2024-07-02 14:43:53.683 sensortag.kexe[35961:336063] I/Kable: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F Connecting
[V/SensorTag] Attempting connection
[I/SensorTag] Connecting...
2024-07-02 14:43:53.998 sensortag.kexe[35961:336063] D/Kable: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F CentralManagerDelegate state change
  state: DidConnect(identifier=6A9BBECA-EC4D-5F75-4DE5-2F59E392836F)
2024-07-02 14:43:53.998 sensortag.kexe[35961:336063] V/Kable: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F discoverServices
2024-07-02 14:43:54.868 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverServices
2024-07-02 14:43:55.048 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverCharacteristicsForService
  service: 180A
2024-07-02 14:43:55.049 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: 0000180a-0000-1000-8000-00805f9b34fb
  characteristic: 00002a23-0000-1000-8000-00805f9b34fb
2024-07-02 14:43:55.050 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: 0000180a-0000-1000-8000-00805f9b34fb
  characteristic: 00002a24-0000-1000-8000-00805f9b34fb
2024-07-02 14:43:55.052 sensortag.kexe[35961:336070] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: 0000180a-0000-1000-8000-00805f9b34fb
  characteristic: 00002a25-0000-1000-8000-00805f9b34fb
2024-07-02 14:43:55.053 sensortag.kexe[35961:336070] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: 0000180a-0000-1000-8000-00805f9b34fb
  characteristic: 00002a26-0000-1000-8000-00805f9b34fb
2024-07-02 14:43:55.054 sensortag.kexe[35961:336070] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: 0000180a-0000-1000-8000-00805f9b34fb
  characteristic: 00002a27-0000-1000-8000-00805f9b34fb
2024-07-02 14:43:55.055 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: 0000180a-0000-1000-8000-00805f9b34fb
  characteristic: 00002a28-0000-1000-8000-00805f9b34fb
2024-07-02 14:43:55.056 sensortag.kexe[35961:336070] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: 0000180a-0000-1000-8000-00805f9b34fb
  characteristic: 00002a29-0000-1000-8000-00805f9b34fb
2024-07-02 14:43:55.057 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: 0000180a-0000-1000-8000-00805f9b34fb
  characteristic: 00002a2a-0000-1000-8000-00805f9b34fb
2024-07-02 14:43:55.058 sensortag.kexe[35961:336070] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: 0000180a-0000-1000-8000-00805f9b34fb
  characteristic: 00002a50-0000-1000-8000-00805f9b34fb
2024-07-02 14:43:55.168 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverCharacteristicsForService
  service: 180F
2024-07-02 14:43:55.258 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: 0000180f-0000-1000-8000-00805f9b34fb
  characteristic: 00002a19-0000-1000-8000-00805f9b34fb
2024-07-02 14:43:55.437 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverCharacteristicsForService
  service: F000AA00-0451-4000-B000-000000000000
2024-07-02 14:43:55.500 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000aa00-0451-4000-b000-000000000000
  characteristic: f000aa01-0451-4000-b000-000000000000
2024-07-02 14:43:55.502 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000aa00-0451-4000-b000-000000000000
  characteristic: f000aa02-0451-4000-b000-000000000000
2024-07-02 14:43:55.503 sensortag.kexe[35961:336070] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000aa00-0451-4000-b000-000000000000
  characteristic: f000aa03-0451-4000-b000-000000000000
2024-07-02 14:43:55.682 sensortag.kexe[35961:336070] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverCharacteristicsForService
  service: F000AA20-0451-4000-B000-000000000000
2024-07-02 14:43:55.737 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000aa20-0451-4000-b000-000000000000
  characteristic: f000aa21-0451-4000-b000-000000000000
2024-07-02 14:43:55.738 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000aa20-0451-4000-b000-000000000000
  characteristic: f000aa22-0451-4000-b000-000000000000
2024-07-02 14:43:55.740 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000aa20-0451-4000-b000-000000000000
  characteristic: f000aa23-0451-4000-b000-000000000000
2024-07-02 14:43:55.918 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverCharacteristicsForService
  service: F000AA40-0451-4000-B000-000000000000
2024-07-02 14:43:55.978 sensortag.kexe[35961:336070] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000aa40-0451-4000-b000-000000000000
  characteristic: f000aa41-0451-4000-b000-000000000000
2024-07-02 14:43:55.978 sensortag.kexe[35961:336070] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000aa40-0451-4000-b000-000000000000
  characteristic: f000aa42-0451-4000-b000-000000000000
2024-07-02 14:43:55.979 sensortag.kexe[35961:336070] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000aa40-0451-4000-b000-000000000000
  characteristic: f000aa44-0451-4000-b000-000000000000
2024-07-02 14:43:56.158 sensortag.kexe[35961:336070] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverCharacteristicsForService
  service: F000AA80-0451-4000-B000-000000000000
2024-07-02 14:43:56.217 sensortag.kexe[35961:336070] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000aa80-0451-4000-b000-000000000000
  characteristic: f000aa81-0451-4000-b000-000000000000
2024-07-02 14:43:56.218 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000aa80-0451-4000-b000-000000000000
  characteristic: f000aa82-0451-4000-b000-000000000000
2024-07-02 14:43:56.219 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000aa80-0451-4000-b000-000000000000
  characteristic: f000aa83-0451-4000-b000-000000000000
2024-07-02 14:43:56.428 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverCharacteristicsForService
  service: F000AA70-0451-4000-B000-000000000000
2024-07-02 14:43:56.493 sensortag.kexe[35961:336070] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000aa70-0451-4000-b000-000000000000
  characteristic: f000aa71-0451-4000-b000-000000000000
2024-07-02 14:43:56.496 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000aa70-0451-4000-b000-000000000000
  characteristic: f000aa72-0451-4000-b000-000000000000
2024-07-02 14:43:56.496 sensortag.kexe[35961:336070] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000aa70-0451-4000-b000-000000000000
  characteristic: f000aa73-0451-4000-b000-000000000000
2024-07-02 14:43:56.608 sensortag.kexe[35961:336070] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverCharacteristicsForService
  service: FFE0
2024-07-02 14:43:56.668 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: 0000ffe0-0000-1000-8000-00805f9b34fb
  characteristic: 0000ffe1-0000-1000-8000-00805f9b34fb
2024-07-02 14:43:56.818 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverCharacteristicsForService
  service: F000AA64-0451-4000-B000-000000000000
2024-07-02 14:43:56.819 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000aa64-0451-4000-b000-000000000000
  characteristic: f000aa65-0451-4000-b000-000000000000
2024-07-02 14:43:56.820 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000aa64-0451-4000-b000-000000000000
  characteristic: f000aa66-0451-4000-b000-000000000000
2024-07-02 14:43:56.999 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverCharacteristicsForService
  service: F000AC00-0451-4000-B000-000000000000
2024-07-02 14:43:57.001 sensortag.kexe[35961:336070] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000ac00-0451-4000-b000-000000000000
  characteristic: f000ac01-0451-4000-b000-000000000000
2024-07-02 14:43:57.002 sensortag.kexe[35961:336070] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000ac00-0451-4000-b000-000000000000
  characteristic: f000ac02-0451-4000-b000-000000000000
2024-07-02 14:43:57.003 sensortag.kexe[35961:336070] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000ac00-0451-4000-b000-000000000000
  characteristic: f000ac03-0451-4000-b000-000000000000
2024-07-02 14:43:57.179 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverCharacteristicsForService
  service: F000CCC0-0451-4000-B000-000000000000
2024-07-02 14:43:57.238 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000ccc0-0451-4000-b000-000000000000
  characteristic: f000ccc1-0451-4000-b000-000000000000
2024-07-02 14:43:57.239 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000ccc0-0451-4000-b000-000000000000
  characteristic: f000ccc2-0451-4000-b000-000000000000
2024-07-02 14:43:57.240 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000ccc0-0451-4000-b000-000000000000
  characteristic: f000ccc3-0451-4000-b000-000000000000
2024-07-02 14:43:57.568 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverCharacteristicsForService
  service: F000FFC0-0451-4000-B000-000000000000
2024-07-02 14:43:57.628 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000ffc0-0451-4000-b000-000000000000
  characteristic: f000ffc1-0451-4000-b000-000000000000
2024-07-02 14:43:57.689 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000ffc0-0451-4000-b000-000000000000
  characteristic: f000ffc2-0451-4000-b000-000000000000
2024-07-02 14:43:57.748 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000ffc0-0451-4000-b000-000000000000
  characteristic: f000ffc3-0451-4000-b000-000000000000
2024-07-02 14:43:57.867 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didDiscoverDescriptorsForCharacteristic
  service: f000ffc0-0451-4000-b000-000000000000
  characteristic: f000ffc4-0451-4000-b000-000000000000
2024-07-02 14:43:57.871 sensortag.kexe[35961:336063] V/Kable: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F Configuring characteristic observations
2024-07-02 14:43:57.871 sensortag.kexe[35961:336063] D/Kable: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F CentralManager.notify
  service: f000aa80-0451-4000-b000-000000000000
  characteristic: f000aa81-0451-4000-b000-000000000000
2024-07-02 14:43:57.927 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didUpdateNotificationStateForCharacteristic
  service: f000aa80-0451-4000-b000-000000000000
  characteristic: f000aa81-0451-4000-b000-000000000000
2024-07-02 14:43:57.928 sensortag.kexe[35961:336063] I/Kable: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F Connected
[I/SensorTag] Connected
[V/SensorTag] Writing gyro period
2024-07-02 14:43:57.928 sensortag.kexe[35961:336063] D/Kable: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F write
  service: f000aa80-0451-4000-b000-000000000000
  characteristic: f000aa83-0451-4000-b000-000000000000
  writeType: WithResponse
  data: FF
2024-07-02 14:43:57.991 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didWriteValueForCharacteristic
  service: f000aa80-0451-4000-b000-000000000000
  characteristic: f000aa83-0451-4000-b000-000000000000
[I/SensorTag] Writing gyro period complete
[I/SensorTag] Enabling gyro
2024-07-02 14:43:57.992 sensortag.kexe[35961:336063] D/Kable: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F write
  service: f000aa80-0451-4000-b000-000000000000
  characteristic: f000aa82-0451-4000-b000-000000000000
  writeType: WithResponse
  data: 7F 00
2024-07-02 14:43:58.047 sensortag.kexe[35961:336071] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didWriteValueForCharacteristic
  service: f000aa80-0451-4000-b000-000000000000
[I/SensorTag] Gyro enabled
  characteristic: f000aa82-0451-4000-b000-000000000000
[V/SensorTag] Waiting to reconnect
2024-07-02 14:43:58.287 sensortag.kexe[35961:336070] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didUpdateValueForCharacteristic
  service: f000aa80-0451-4000-b000-000000000000
  characteristic: f000aa81-0451-4000-b000-000000000000
  data: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
[I/SensorTag] Vector3f(x=0.0, y=0.0, z=0.0)
[I/SensorTag] Received state: Connected
2024-07-02 14:44:00.838 sensortag.kexe[35961:336070] D/Kable/Delegate: 6A9BBECA-EC4D-5F75-4DE5-2F59E392836F didUpdateValueForCharacteristic
  service: f000aa80-0451-4000-b000-000000000000
[I/SensorTag] Vector3f(x=-30.776978, y=-54.122925, z=34.38568)
```
</details>

**Comment by twyatt** [comment]
Thanks for the contribution!

**Pull request merged**
